### PR TITLE
Feature/lvgl v8.2.0

### PR DESCRIPTION
**Pull request by CvRXX**
This PR bumps LVGL to the v8.2.0 version. The functionality should be exactly the same as on `develop`. 

The biggest change in code is that the layout of the blocks is now done with the new grid feature which removes a lot of boilerplate.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -54,8 +55,10 @@ RUN : \
         usbutils \
         xz-utils \
         zip \
+        nodejs \
```
Comment (steersbob) [inline .devcontainer/Dockerfile:58]: this list is sorted alphabetically

File: app/brewblox-esp/sdkconfig [hunk 2]
```diff
@@ -1322,6 +1330,297 @@ CONFIG_WPA_MBEDTLS_CRYPTO=y
 # CONFIG_WPA_WPS_STRICT is not set
 # CONFIG_WPA_11KV_SUPPORT is not set
 # end of Supplicant
+
```
Comment (steersbob) [inline app/brewblox-esp/sdkconfig:1333]: Looks like lvgl snuck into sdkconfig again. Is it intentional this time?

File: platform/all/graphics/fonts/generate_fonts.py [hunk 3]
```diff
@@ -40,31 +40,27 @@
 size_main = 16
 size_main_small = 12
 
-if not which("npx"):
-    print("npx not found, to install, run 'npm i -g npx'")
-    exit(1)
-
 os.system(
```
Comment (steersbob) [inline platform/all/graphics/fonts/generate_fonts.py:43]: Typically, `subprocess.run([*args])` is preferable to os.system, but this is not critical.

File: .devcontainer/Dockerfile [hunk 4]
```diff
@@ -54,8 +55,10 @@ RUN : \
...
     && update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 20 \
     && update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-10 20 \
+    && npm i lv_font_conv -g \
```
Comment (steersbob) [inline .devcontainer/Dockerfile:61]: Put installed packages on an indented new line. This significantly increases readability of both the Dockerfile itself, and future diffs.
```
&& npm install -g \
    lv_font_conv \
```

**Comment by steersbob** [comment]
Can't say much about the C++ side of things, so mostly notes to keep config clean.

**Review by steersbob — CHANGES_REQUESTED** [review]

File: .devcontainer/Dockerfile (hunk 4 above)
Comment (CvRXX) [reply .devcontainer/Dockerfile]: fixed in 70720752a92e85ad1286e4f6bff4dab053b3d1e0

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (CvRXX) [reply .devcontainer/Dockerfile]: fixed in 70720752a92e85ad1286e4f6bff4dab053b3d1e0

File: app/brewblox-esp/sdkconfig (hunk 2 above)
Comment (CvRXX) [reply app/brewblox-esp/sdkconfig]: fixed in 0c6fc0536ff469cfedc4f494f5bdfbcf6d57e38f

File: platform/all/graphics/layout.cpp [hunk 5]
```diff
@@ -59,36 +71,36 @@ void Layout::updateConfig()
         tzset();
 
         auto makeWidget = [&settings, this](uint8_t pos) -> std::unique_ptr<BaseWidget> {
```
Comment (elcojacobs) [inline platform/all/graphics/layout.cpp:73]: lamda is constructed and destructed. Maybe make static const and pass captures as args?

File: platform/all/graphics/widgets/BaseWidget.cpp [hunk 6]
```diff
@@ -5,26 +5,31 @@ uint8_t luminance(const lv_color_t& color)
...
 
     // set styling of object and its children based on luminance of background color
-    lv_obj_add_style(obj, LV_OBJ_PART_MAIN, luminance(color) < 128 ? &style::bg_dark : &style::bg_light);
+    lv_obj_add_style(obj, luminance(color) < 128 ? &style::bg_dark : &style::bg_light, 0);
 
-    lv_obj_add_style(label, LV_LABEL_PART_MAIN, &style::widget_name);
+    lv_obj_add_style(label, &style::widget_name, 0);
     lv_label_set_text(label, labelText);
-    lv_obj_align(label, nullptr, LV_ALIGN_IN_BOTTOM_MID, 0, -4);
+    lv_obj_align(label, LV_ALIGN_BOTTOM_MID, 0, 4);
```
Comment (elcojacobs) [inline platform/all/graphics/widgets/BaseWidget.cpp:27]: is -4 to 4 intentional?

**Review by elcojacobs — APPROVED** [review]

File: platform/all/graphics/widgets/BaseWidget.cpp (hunk 6 above)
Comment (CvRXX) [reply platform/all/graphics/widgets/BaseWidget.cpp:27]: Yes, it is, I think the alignment functions changed.

**Review by elcojacobs — CHANGES_REQUESTED** [review]
trailing return type on function not in line with rest of code base

**Pull request merged**